### PR TITLE
Switch to arm64/aarch64 electron on macOS M1

### LINKED ISSUE #1416: Switch to arm64/aarch64 electron on macOS M1
### Rancher Desktop Version

1.0.0

### Rancher Desktop K8s Version

?

### What operating system are you using?

macOS

### Operating System / Build Version

macos Monterey 12.1

### What CPU architecture are you using?

arm64 (Apple Silicon)

### Linux only: what package format did you use to install Rancher Desktop?

_No response_

### Windows User Only

_No response_

### Actual Behavior

When opening rancher-desktop it still requires to install Rosetta 2

### Steps to Reproduce

1. Download rancher desktop from: https://github.com/rancher-sandbox/rancher-desktop/releases/download/v1.0.0/Rancher.Desktop-1.0.0.aarch64.dmg
1. Copy the app to the Applications folder
1. Double click to run the app without having Rosetta 2 installed

### Result

You get the Rosetta 2 installation dialog

### Expected Behavior

Should work without requiring Rosetta 2

### Additional Information

Related to https://github.com/rancher-sandbox/rancher-desktop/issues/16

### PR DESCRIPTION
**Pull request by jsoref**
closes #1416

### CODE REVIEW

**Comment by jandubois** [comment]
@jsoref Please fix the linter errors so we can get a proper CI run for this PR!

**Review by jandubois — CHANGES_REQUESTED** [review]
Please fix linter errors!

**Comment by sanglt** [comment]
Hi @jsoref do you have any update from the draft MR? Can I continue working on your branch because I don't want to install Rosetta on my mac.

**Comment by jsoref** [comment]
I'm distracted, you're welcome to try to push it forward. I don't expect to have time to focus on this for a while 😦

**Comment by douglas** [comment]
@jandubois Is it still relevant? I have some time and I would love to contribute to Rancher Desktop.

**Comment by jandubois** [comment]
> Note that we are going to bump both Electron and Node versions right after the 1.11 release, so this might affect how the app needs to be build. Maybe wait until that is done (should be within another week or so), and then take a look.

@douglas 1.11 has just been released and the Electron and Node bumps have been merged into `main`, so you should be good to go and try things...

**Comment by douglas** [comment]
> > Note that we are going to bump both Electron and Node versions right after the 1.11 release, so this might affect how the app needs to be build. Maybe wait until that is done (should be within another week or so), and then take a look.
> 
> @douglas 1.11 has just been released and the Electron and Node bumps have been merged into `main`, so you should be good to go and try things...

Nice!! I'll schedule some time next week to work on it 👍

File: scripts/download/lima.mjs [hunk 1]
```diff
@@ -52,15 +52,20 @@ async function getAlpineLima(arch) {
...
   if (platform === 'darwin') {
     platform = 'macos';
-    if (process.env.M1) {
-      arch = 'aarch64';
-      platform = `macos-${ arch }`;
+    if (process.env.M1 !== '0' && (process.env.M1 || process.arch === 'arm64')) {
+      const m1arch = 'aarch64';
+      const m1platform = `macos-${ m1arch }`;
+
+      promises.push([getLima(m1platform), getAlpineLima(m1arch, m1platform)]);
```
Comment (matthieusieben) [inline scripts/download/lima.mjs:65]: You wan to push pomises, not array of promises:
```suggestion
      promises.push(getLima(m1platform), getAlpineLima(m1arch, m1platform));
```

File: scripts/download/lima.mjs [hunk 2]
```diff
@@ -52,15 +52,20 @@ async function getAlpineLima(arch) {
...
     }
   }
+  promises.push([getLima(platform), getAlpineLima(arch, platform)]);
```
Comment (matthieusieben) [inline scripts/download/lima.mjs:68]: Same here:
```suggestion
  promises.push(getLima(platform), getAlpineLima(arch, platform));
```

File: scripts/download/lima.mjs (hunk 1 above)
Comment (jsoref) [reply scripts/download/lima.mjs:65]: Sadly the code has been refactored (`Lima` is now a class ... in typescript).

Offhand, I suspect I want to move some of this code to scripts/postinstall.ts, but I don't have the energy now...

**Comment by jandubois** [comment]
This functionality has been implemented via #6255 and #6264. Once those are merged the only remaining executable requiring Rosetta will be `kuberlr` (and the `kubectl` managed by `kuberlr`).

**Pull request closed without merging**